### PR TITLE
Add a new Calendar advanced service example.

### PR DESCRIPTION
**Pull request by ryan-devrel**
Adds a sample to the Calendar Advanced Service set. The sample covers using sync tokens when retrieving events, similar to the Calendar API code listed here:

https://developers.google.com/calendar/v3/sync#sample_code

### CODE REVIEW

File: advanced/calendar.gs [hunk 1]
```diff
@@ -129,3 +129,72 @@ function parseDate(string) {
...
+  var pageToken;
+  var options = {
+    maxResults: 100
+  };
+  var syncToken = properties.getProperty('syncToken');
+  if (syncToken && !fullSync) {
+    options['syncToken'] = syncToken;
+  } else {
+    // Sync events up to thirty days in the past.
+    options['timeMin'] = getRelativeDate(-30,0).toISOString();
```
Comment (erickoledadevrel) [inline advanced/calendar.gs]: Nitpick: Add a space after "-30,"

File: advanced/calendar.gs [hunk 2]
```diff
@@ -129,3 +129,72 @@ function parseDate(string) {
...
+      events = Calendar.Events.list(calendarId, options);
+    } catch (e) {
+      // Check to see if the sync token was invalidated by the server;
+      // if so, perform a full sync instead.
+      if (e.message === "Sync token is no longer valid, a full sync is required." ) {
+        properties.deleteProperty('syncToken');
+        logSyncedEvents(calendarId, true);
+        return;
+      } else {
+        throw e.message;
```
Comment (erickoledadevrel) [inline advanced/calendar.gs]: Nitpick: Use `throw new Error(e.message)` instead.

File: advanced/calendar.gs [hunk 3]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+ * since the last sync. If the sync token is missing or invalid, log all
+ * events from up to a month ago (a full sync).
+ *
+ * @param {string} calendarId The ID of the calender to retrieve events from.
+ * @param {boolean} fullSync If true, throw out any existing sync token and
+ *        perform a full sync; if false, use the existing sync token if possible.
+ */
+function logSyncedEvents(calendarId, fullSync) {
+  var properties = PropertiesService.getUserProperties();
+  var pageToken;
```
Comment (grant) [inline advanced/calendar.gs]: Move this var after `var events;`
It is not used until the do while loop.

File: advanced/calendar.gs [hunk 4]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+  var options = {
+    maxResults: 100
+  };
+  var syncToken = properties.getProperty('syncToken');
+  if (syncToken && !fullSync) {
+    options['syncToken'] = syncToken;
```
Comment (grant) [inline advanced/calendar.gs]: Don't use brackets to access object keys if possible:
`options.syncToken = syncToken;`

File: advanced/calendar.gs [hunk 5]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+  } else {
+    // Sync events up to thirty days in the past.
+    options['timeMin'] = getRelativeDate(-30, 0).toISOString();
```
Comment (grant) [inline advanced/calendar.gs]: `options.timeMin = getRelativeDate(-30, 0).toISOString();`

File: advanced/calendar.gs [hunk 6]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+  }
+
+  // Retrieve events, one page at a time.
```
Comment (grant) [inline advanced/calendar.gs]: I prefer reading this sentence without the comma:
`Retrieve events one page at a time.`

File: advanced/calendar.gs [hunk 7]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+  var events;
+  do {
+    try {
+      options['pageToken'] = pageToken;
```
Comment (grant) [inline advanced/calendar.gs]: `options.pageToken = pageToken;`

**Comment by ryan-devrel** [comment]
Resolved CLA issue correctly this time.

File: advanced/calendar.gs [hunk 8]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+      events = Calendar.Events.list(calendarId, options);
+    } catch (e) {
+      // Check to see if the sync token was invalidated by the server;
+      // if so, perform a full sync instead.
+      if (e.message === "Sync token is no longer valid, a full sync is required." ) {
```
Comment (grant) [inline advanced/calendar.gs]: Remove extra space before end paren.
`" ) {` =>
`") {`

File: advanced/calendar.gs [hunk 9]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+        properties.deleteProperty('syncToken');
+        logSyncedEvents(calendarId, true);
+        return;
+      } else {
+        throw new Error(e.message);
+      }
+    }
+    
+    if (events.items && events.items.length > 0) {
+      for(var i = 0; i < events.items.length; i++) {
```
Comment (grant) [inline advanced/calendar.gs]: `for (`

File: advanced/calendar.gs [hunk 10]
```diff
@@ -129,3 +129,70 @@ function parseDate(string) {
...
+         var event = events.items[i];
+         if (event.status === 'cancelled') {
+           console.log('Event id %s was cancelled.', event.id);
+         } else if (event.start.date) {
+           // All-day event.
+           var start = parseDate(event.start.date);
+           console.log('%s (%s)', event.summary, start.toLocaleDateString());
+         } else {
```
Comment (grant) [inline advanced/calendar.gs]: A comment here stating which case this `else` handles would be nice.
You said "All-day event." above, which was helpful.

**Comment by ryan-devrel** [comment]
Accidentally tried a 'git pull' in there, but hopefully it's correct now.

**Comment by grant** [comment]
@ryan-devrel Can you fix the CI/Travis errors before I can merge?
```
/home/travis/build/gsuitedevs/apps-script-samples/advanced/calendar.gs
  185:2  error  Unexpected tab character  no-tabs
  185:2  error  Mixed spaces and tabs     no-mixed-spaces-and-tabs
```

This resulted in an x for `continuous-integration/travis-ci/pr` > Details.

**Pull request merged**